### DISCUSSION TITLE
Can I set Files On-Demand for selected folders?

### DISCUSSION THREAD

**Discussion by itsMohitShah** [abraunegg/onedrive#2284] [Q&A] [answered]
Hi @abraunegg!
I have a OneDrive license provided by my university where I back up all of my files. 
I have a Windows and Linux dual boot system. As I was following through your repositories, you mentioned that in order to access my OneDrive files from Linux, I need to turn the Files On-Demand off.
However, there are some files and datasets that are taking up so much space that my physical hard drive is almost full. I can save about 300GB of space if I could just make those particular files online. As far as I researched online, I can't have particular files backed up on a completely online basis without turning on the Files On-Demand setting for the entire system. Is there any way where I can save those 300GB worth of data and still access the rest of my files from Ubuntu?
As much as I can tell, this is leaning more towards OneDrive's issue rather than the Linux Client, but you are the best source of reliable information I can think of, hence I am asking you :)

**Comment by rrodrigueznt** [comment]
@itsMohitShah I hope you don't mind I jump into this thread, although your greet @abraunegg! Just a couple of ideas to check my own understanding of the OneDrive ecosystem.

> Hi @abraunegg! I have a OneDrive license provided by my university where I back up all of my files.

I would say it is not a good idea to use OneDrive as a back-up system in the traditional sense. OneDrive, as many others cloud file systems, are designed to replicate changes in all your documents and make them available to you anywhere, any time.

> I have a Windows and Linux dual boot system. As I was following through your repositories, you mentioned that in order to access my OneDrive files from Linux, I need to turn the Files On-Demand off.

Please, could you provide a link to the document/thread where we can find this recommendation?

> However, there are some files and datasets that are taking up so much space that my physical hard drive is almost full. I can save about 300GB of space if I could just make those particular files online. As far as I researched online, I can't have particular files backed up on a completely online basis without turning on the Files On-Demand setting for the entire system. Is there any way where I can save those 300GB worth of data and still access the rest of my files from Ubuntu? As much as I can tell, this is leaning more towards OneDrive's issue rather than the Linux Client, but you are the best source of reliable information I can think of, hence I am asking you :)

Yes, it is possible. You can synchronize want you want, or almost what you want, in your Linux system by using @abraunegg great OneDrive client. Please, check [sync_list](https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#performing-a-selective-sync-via-sync_list-file) under **Configuration and Usage of the OneDrive Free Client**.

**Comment by itsMohitShah** [comment]
> Please, could you provide a link to the document/thread where we can find this recommendation?

You can refer to this link: https://github.com/abraunegg/onedrive/blob/master/docs/advanced-usage.md#configuring-the-client-for-use-in-dual-boot-windows--linux-situations


> Yes, it is possible. You can synchronize want you want, or almost what you want, in your Linux system by using @abraunegg great OneDrive client. Please, check [sync_list](https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#performing-a-selective-sync-via-sync_list-file) under Configuration and Usage of the OneDrive Free Client.

I actually don't need OneDrive to sync in it, just simple access as if it's a part of my local storage will fulfil my needs. I have VERY limited space on my Ubuntu, so I try not to add too much stuff on it. Should I still download the OneDrive Client?

**Comment by abraunegg** [comment, marked as answer]
@itsMohitShah 
> I have a OneDrive license provided by my university where I back up all of my files.

OK .. so the account will be a OneDrive Business Account with SharePoint for any Shared Files & Libraries. This is important.

> I have a Windows and Linux dual boot system. As I was following through your repositories, you mentioned that in order to access my OneDrive files from Linux, I need to turn the Files On-Demand off.

Correct - because Windows, when using the 'on-demand' feature, the files are not actually files, they are reparse points, which cannot be read or understood natively by Linux.

> However, there are some files and datasets that are taking up so much space that my physical hard drive is almost full. I can save about 300GB of space if I could just make those particular files online. As far as I researched online, I can't have particular files backed up on a completely online basis without turning on the Files On-Demand setting for the entire system. Is there any way where I can save those 300GB worth of data and still access the rest of my files from Ubuntu?

Windows does not have the ability to selectivly sync data, unlike this client. When you turn off 'on-demand' on Windows, it will sync all your data down to your system. Nothing can be done to change that.

This being said you have a couple of options:

1. Get a larger drive for your system. Here in Australia you can get a 1TB SSD for $89, with 2TB around $188.
2. Shift your 'least-used' data (archive data or such) out of your main OneDrive Account and into your own SharePoint Library - this way reducing all your files on your OneDrive Account. Change the account settings to not use 'on-demand' so you can access your files in a dual-boot scenario.
3. Use 'onedriver' which supports using 'on-demand' access for OneDrive, native file system that only provides the OneDrive 'on-demand' functionality, open source and free. Supports Personal, Business account types. Currently does not support Shared Folders or SharePoint.

For further details please read:
* https://github.com/abraunegg/onedrive/issues/757
* https://github.com/jstaf/onedriver

**Reply by itsMohitShah** [reply]
Thank you so much for the help!
I really appreciate it
